src/dataiterators/minibatches.jl

```diff
@@ -1,17 +1,16 @@
-default_partitionsize(A) = 20
+default_partitionsize(n) = clamp(div(n,5), 1, 20)
 
 """
 Helper function to compute sensible and compatible values for the
 `size` and `count`
 """
-function _compute_partitionsettings(features, size::Int, count::Int)
-    num_observations = nobs(features)::Int
+function _compute_partitionsettings(num_observations::Int, size::Int, count::Int)
     @assert num_observations > 0
     @assert size  <= num_observations
     @assert count <= num_observations
     if size < 0 && count < 0
         # no batch settings specified, use default size and as many batches as possible
-        size = default_partitionsize(features)::Int
+        size = default_partitionsize(num_observations)::Int
```

Comment:
Mhm, this was originally intended to give a user the possible entry point of controlling the default value for his/her custom datatype. Should we try to preserve that? My reason for that was that it is hard to know what kind of twisted way the data is loaded from; could be from a database, or network, or anything. So maybe one would want to define a sensible default partition for such a special case.